src/components/PacksReadme/PacksReadme.tsx

```diff
@@ -93,15 +93,24 @@ export default function PacksReadme() {
   }, [packName]);
   useEffect(() => {
     const searchParams = window ? new URLSearchParams(window.location.search) : null;
-    const urlParamVersion = searchParams?.get("versions");
+    const urlParamVersion = searchParams?.get("version");
     const version = urlParamVersion || packData?.latestVersion || packData?.versions[0]?.title || "";
-    if (!urlParamVersion) {
-      history.replace({ search: `?pack=${packName}&versions=${version}` });
-    }
+    let packDataObj;
     if (version && !version.endsWith(".x")) {
-      const tag = packData?.versions.find((tagVersion) => tagVersion.title === version.replace(/.\d+$/, ".x"))
-      const queryParamVersionData = tag && tag.children.find((child: any) => child.title === version);
-      setSelectedPackUid(queryParamVersionData?.packUid || "");
+      packDataObj = packData?.versions.reduce((acc, tagVersion) => {
+        let childVersion;
+        if (childVersion = tagVersion.children.find((child:any) => child.title === version)) {
+          acc = {
+            childPackUid: childVersion.packUid,
+            parentVersion: tagVersion.title,
+          }
+        }
+        return acc;
+      }, {});
+      setSelectedPackUid(packDataObj.childPackUid || "");
+    }
+    if (!urlParamVersion) {
+      history.replace({ search: `?pack=${packName}&version=${version}&parent=${packDataObj?.parentVersion || ""}` });
```

Comment:
If you send the parentVersion from the beginning this entire logic would not be needed

Reply:
see comments above